amazon/ion/writer_binary_raw.py

```diff
@@ -263,25 +253,39 @@ def _serialize_timestamp(ion_event):
     if precision.includes_second:
         length += _write_varuint(value_buf, dt.second)
-        coefficient = dt.microsecond
+        coefficient_fraction_seconds = getattr(ion_event.value, TIMESTAMP_FRACTIONAL_SECONDS_FIELD, None)
         fractional_precision = getattr(ion_event.value, TIMESTAMP_FRACTION_PRECISION_FIELD, MICROSECOND_PRECISION)
-        if coefficient is not None and fractional_precision is not None:
+        coefficient = dt.microsecond
+        if coefficient is not None and fractional_precision is not None and coefficient_fraction_seconds is None:
             if coefficient == 0:
                 adjusted_fractional_precision = fractional_precision
             else:
                 adjusted_fractional_precision = MICROSECOND_PRECISION
+                if fractional_precision > MICROSECOND_PRECISION:
+                    adjusted_fractional_precision = fractional_precision
                 # This optimizes the size of the fractional encoding when the extra precision is not needed.
                 while adjusted_fractional_precision > fractional_precision and coefficient % 10 == 0:
                     coefficient //= 10
                     adjusted_fractional_precision -= 1
-            if adjusted_fractional_precision > fractional_precision or \
-                    coefficient > _TEN_EXP_MINUS_ONE[fractional_precision]:
+            if adjusted_fractional_precision > fractional_precision:
                 raise ValueError('Error writing event %s. Found timestamp fractional precision of %d digits, '
                                  'which is less than needed to serialize %d microseconds.'
                                  % (ion_event, fractional_precision, dt.microsecond))
             exponent = -adjusted_fractional_precision
             if not (coefficient == 0 and exponent >= 0):
                 length += _write_decimal_value(value_buf, exponent, coefficient)
+        else:
+            if coefficient_fraction_seconds is not None:
+                exponent = -fractional_precision
+                coefficient_fraction_seconds_decimal = (Decimal(coefficient_fraction_seconds *
+                                                                (10**fractional_precision)))
+                coefficient_fraction_seconds_int = int(Decimal(coefficient_fraction_seconds *
+                                                               (10 ** fractional_precision)))
+                if coefficient_fraction_seconds_decimal > coefficient_fraction_seconds_int:
+                    raise ValueError('Error writing event %s. Found timestamp fractional precision of %d digits, '
+                                     'which is less than needed to serialize %d microseconds.'
+                                     % (ion_event, fractional_precision, coefficient_fraction_seconds_decimal))
+                length += _write_decimal_value(value_buf, exponent, coefficient_fraction_seconds_int)
```

Comment:
I think you can factor writer_binary_raw's `_serialize_decimal` lines [177-186](https://github.com/amzn/ion-python/blob/master/amazon/ion/writer_binary_raw.py#L177) into a method and reuse it here. That way you can just pass it the decimal without performing any math here.

Reply:
Done.